### PR TITLE
fix:  fix the order of layerState when AnimatorStatePlayState.Finished

### PR DESCRIPTION
**Pull request by luzhuang**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: packages/core/src/animation/Animator.ts [hunk 1]
```diff
@@ -473,11 +473,16 @@ export class Animator extends Component {
...
+    if (playState === AnimatorStatePlayState.Finished) {
+      layerData.layerState = LayerState.Standby;
+    }
+
     if (lastPlayState === AnimatorStatePlayState.UnStarted) {
       this._callAnimatorScriptOnEnter(state, layerIndex);
     }
     if (playState === AnimatorStatePlayState.Finished) {
       this._callAnimatorScriptOnExit(state, layerIndex);
+      return;
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:485]: If add rerturn here, The last frame maybe not update?

File: packages/core/src/animation/Animator.ts [hunk 2]
```diff
@@ -493,9 +484,19 @@ export class Animator extends Component {
     }
     playData.frameTime += state.speed * delta;
 
+
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/animation/Animator.ts:487]: Delete this line.

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**